### PR TITLE
Bluetooth: controller: split: ULL filter API modifications

### PR DESCRIPTION
**Pull request by erbr-ot**
Moving struct definitions into header file and adding API to
allow accessing data-structures from lll context

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/lll_filter.h [hunk 1]
```diff
@@ -13,12 +13,43 @@ struct lll_filter {
 	u8_t  bdaddr[WL_SIZE][BDADDR_SIZE];
 };
 
+/* Whitelist peer list */
+struct white_list {
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/lll_filter.h]: I think the structures should have `lll_filter` as prefix to follow general naming conventions.

**Review by wopu-ot — CHANGES_REQUESTED** [review]
The commit title does not follow the conventions either.

File: subsys/bluetooth/controller/ll_sw/lll_filter.h (hunk 1 above)
Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/lll_filter.h]: Good point - however - shouldn't it be for instance lll_whitelist? To be in line with struct lll_filter {}

Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/lll_filter.h]: There is a bit of mess here, naming wise - ULL vs. LLL I mean? In a way they should be named ull_whitelist/resolvelist, as they are 'at home' in ull_filter, on the other hand lll_filter.h was used to hold the ull_....() API functions and lll_filter is still called lll_filter.

**Review by wopu-ot — APPROVED** [review]
LGTM

File: subsys/bluetooth/controller/ll_sw/ull_filter.c [hunk 2]
```diff
@@ -748,6 +724,16 @@ struct lll_filter *ull_filter_lll_get(bool whitelist)
 }
 
 #if defined(CONFIG_BT_CTLR_PRIVACY)
+struct lll_whitelist *ull_filter_lll_get_whitelist(void)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_filter.c]: ull_filter_lll_whitelist_get

File: subsys/bluetooth/controller/ll_sw/ull_filter.c [hunk 3]
```diff
@@ -748,6 +724,16 @@ struct lll_filter *ull_filter_lll_get(bool whitelist)
...
+{
+	return wl;
+}
+
+struct lll_resolvelist *ull_filter_lll_get_resolvelist(void)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_filter.c]: ull_filter_lll_resolvelist_get

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_filter.c (hunk 3 above)
Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c]: Yeap

**Review by cvinayak — APPROVED** [review]

**Review by wopu-ot — APPROVED** [review]

**Pull request merged**